### DISCUSSION TITLE
Can make magic link work cross devices?

### DISCUSSION THREAD

**Discussion by shouze** [nextauthjs/next-auth#309] [Help]
*Please refer to the [documentation](https://next-auth.js.org/getting-started/introduction), the [example project](https://github.com/iaincollins/next-auth-example) and existing issues before creating a new issue.*

**Your question**
Can make magic link work cross devices?

**What are you trying to do**
Given passwordless & magic link signin, is it possible to add the feature that kind of scenario:
1. I start signin process on my desktop
2. I open email with magic link on my phone and click on it
3. I get logged in on my desktop as on my phone

**Documentation feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by iaincollins** [comment]
This isn't currently supported, but would welcome something like this as a feature request!

I would love to have session management and sign in / sign out be possible across devices.

**Comment by stale** [comment]
Hi there! It looks like this issue hasn't had any activity for a while. It will be closed if no further activity occurs. If you think your issue is still relevant, feel free to comment on it to keep ot open. Thanks!

**Comment by alaeddine119** [comment]
May I add to the discussion that an another way could be a temporary magic code in addition to a magic link, like the system used by Glitch for example.

![Capture](https://user-images.githubusercontent.com/25179455/103604460-6f94e780-4f11-11eb-8a88-e89f2fa23ce4.PNG)

**Reply by iaincollins** [reply]
Thanks! Agree something like being able to use a link *or* enter a code would be an ideal user experience.

I think the codes should be short for convenience (e.g. ~6 chars) but support that securely we may also need to ensure they cannot be abused, which seems like it might require some changes to the current implementation - and possibly verification schema (and potentially to adaptors, ugh) which is currently what is holding things up.

I'm very keen to see this though!
